### ISSUE TITLE
Changing -n value error BSGS mode

### ISSUE THREAD

**Issue by AliGindahouseGit** [albertobsd/keyhunt#103]
Hello Alberto,

I've noticed when I try to chanke -n value BSGS mode with -K 6000 I get two types of error:

1 - not divisible by 1024
2 - [E] -n param doesn't have exact square root

Thanks,

Ali G

**Comment by albertobsd** [comment]
Thanks for report it, yes I'm solving those bugs right now.

I will notify you when i fix it.

**Comment by Noname400** [comment]
Not work -n
[E] M value is not divisible by 1024
keyhunt -m bsgs -f tests/120.txt -b 120 -R -k 20 -n 1099511627776

**Comment by albertobsd** [comment]
Yes that is a bug, internally the code try to correct some values, but there are not enough validations. The next update will correct that behavior

Some explanation about K, N and M

Usually without setting the K value (Default 1)
M is always root(N) also we need that M was divisible by 1024 to do some tricks with the bP table and the bloom filters, just to save memory.

But when we setup K then the things become a little more complicated, at firts M is  root(N), but then we divide by K
 M= M / K. Also we need to multiply the bP Table items * K, just to compensate the difference, with this we gain speed, less M mean less Operations but more memory usage.

I will try to simplify the things
